test_jpeg/jpegenc/subblocks/RLE/RLECore/rlecore.py

```diff
@@ -0,0 +1,179 @@
+""" core of the run length encoder module """
+
+from myhdl import always_comb, always_seq, block
+from myhdl import intbv, ResetSignal, Signal
+from myhdl.conversion import analyze
+from jpegenc.subblocks.RLE.RLECore.entropycoder import entropycoder
+
+
+WIDTH_RAM_ADDRESS = 6
+WIDTH_RAM_DATA = 12
+SIZE = 4
+
+class DataStream(object):
+    """ Input data streams into Rle Core """
+    def __init__(self, width=WIDTH_RAM_DATA):
+        self.start = Signal(bool(0))
+        self.input_val = Signal(intbv(0)[width:].signed())
+
+class RLESymbols(object):
+    """RLE symbols generatred by RLE Core"""
+    def __init__(self, width=WIDTH_RAM_DATA, size=SIZE):
+        self.runlength = Signal(intbv(0)[size:])
+        self.size = Signal(intbv(0)[size:])
+        self.amplitude = Signal(intbv(0)[width:].signed())
+        self.dovalid = Signal(bool(0))
+
+class RLEConfig(object):
+    """RLE configuration Signals are added here"""
+    def __init__(self):
+        self.color_component = Signal(intbv(0)[3:])
+        self.read_addr = Signal(intbv(0)[6:])
+        self.sof = Signal(bool(0))
+
+def sub(num1, num2):
+    """subtractor for Difference Encoder"""
+    return num1 - num2
+
+@block
+def rle(reset, clock, datastream, rlesymbols, rleconfig):
+    """ Input and start Signal send using datastream
+        and rle symbols are captured
+    """
```

Comment:
The interfaces look reasonable.  This is a good use of interfaces, it simplifies the run-length-encoder block considerably.  It is much easier for someone to look at this and understand what is going on.

You need to expand the doc strings, use the google style that is outlined in the best practices.

```
Arguments:
    datastream: input datastream from ??? < and other useful information >
    rlesymbols: the output encoded symbol stream, to ??? <and other useful information>
    rleconfig: what is the config, is it static (can it be) is it dynamic ...
```

Add more information, should be clear what this block does and some info (but limited) how it does it.